addons/pos_razorpay/static/src/components/payment_screen/payment_screen.js

```diff
@@ -0,0 +1,19 @@
+/** @odoo-module */
+
+import { PaymentScreen } from "@point_of_sale/app/screens/payment_screen/payment_screen";
+import { patch } from "@web/core/utils/patch";
+import { onMounted } from "@odoo/owl";
+
+patch(PaymentScreen.prototype, {
+    setup() {
+        super.setup(...arguments);
+            onMounted(() => {
+                const pendingPaymentLine = this.currentOrder.paymentlines.find(
+                    paymentLine => paymentLine.payment_method.use_payment_terminal === 'razorpay' && (!paymentLine.is_done() && paymentLine.get_payment_status() !== 'pending')
+                );
+                if (pendingPaymentLine) {
+                    pendingPaymentLine.payment_method.payment_terminal._waitForPaymentConfirmation();
+                }
```

Comment:
@qle-odoo This should handle the refresh and we can continue where we left

Reply:
Ok good.

As discussed, it would be wise to use `websocket` communication to report the status of the transaction to the web client instead of a long polling.